tools/@aws-cdk/cli-args-gen/lib/cli-type-gen.ts

```diff
@@ -0,0 +1,140 @@
+  scope.documentation.push('-------------------------------------------------------------------------------------------');
+
+  const cliArgType = new StructType(scope, {
+    export: true,
+    name: 'CliArguments',
+    docs: {
+      summary: 'The structure of the CLI configuration, generated from packages/aws-cdk/lib/config.ts',
+    },
+  });
+
+  // add required command
+  cliArgType.addProperty({
+    name: '_',
+    type: Type.arrayOf(Type.STRING),
+    docs: {
+      summary: 'The CLI command name followed by any properties of the command',
+    },
+  });
+
+  // add global options
+  const globalOptionType = new StructType(scope, {
+    export: true,
+    name: 'GlobalOptions',
+    docs: {
+      summary: 'Global options available to all CLI commands',
+    },
+  });
+  for (const [optionName, option] of Object.entries(config.globalOptions)) {
+    globalOptionType.addProperty({
+      name: optionName,
+      type: convertType(option.type),
+      docs: {
+        default: normalizeDefault(option.default),
+        summary: option.desc,
+        deprecated: option.deprecated ? String(option.deprecated) : undefined,
+      },
+      optional: true,
+    });
+  }
+  cliArgType.addProperty({
+    name: 'globalOptions',
+    type: Type.fromName(scope, globalOptionType.name),
+    docs: {
+      summary: 'Global options available to all CLI commands',
+    },
+    optional: true,
+  });
+
+  // add command-specific options
+  for (const [commandName, command] of Object.entries(config.commands)) {
+    const commandType = new StructType(scope, {
+      export: true,
+      name: `${kebabToPascal(commandName)}Options`,
+      docs: {
+        summary: command.description,
+        remarks: command.aliases ? `aliases: ${command.aliases.join(' ')}` : undefined,
+      },
+    });
+
+    for (const [optionName, option] of Object.entries(command.options ?? {})) {
+      commandType.addProperty({
+        name: optionName,
+        type: convertType(option.type),
+        docs: {
+          default: normalizeDefault(option.default),
+          summary: option.desc,
+          deprecated: option.deprecated ? String(option.deprecated) : undefined,
+          remarks: option.alias ? `aliases: ${Array.isArray(option.alias) ? option.alias.join(' ') : option.alias}` : undefined,
+        },
+        optional: true,
+      });
+    }
+
+    cliArgType.addProperty({
+      name: commandName,
+      type: Type.fromName(scope, commandType.name),
+      docs: {
+        summary: command.description,
+        remarks: command.aliases ? `aliases: ${command.aliases.join(' ')}` : undefined,
+      },
+      optional: true,
+    });
+  }
+
+  const ts = new TypeScriptRenderer({
+    disabledEsLintRules: [
+      EsLintRules.COMMA_DANGLE,
+      EsLintRules.COMMA_SPACING,
+      EsLintRules.MAX_LEN,
+      EsLintRules.QUOTES,
+      EsLintRules.QUOTE_PROPS,
+    ],
+  }).render(scope);
```

Comment:
these are the same disabled eslint rules that the other file uses. i don't love the comma related rules being disabled, and i want to see if i can change them in a separate PR, but super low priority.

Reply:
If you change the prettier formatting below to do the right thing, you might not need to disable some of the rules.